site/docs/expressions/Extension_Functions/index.md

```diff
@@ -0,0 +1,219 @@
+| [Table](table_functions.md)         | Scalar | Vector | TBD                                                                            |
+
+The properties specific to these function types are described in more detail in the linked sections.
+
+!!! note
+
+    Do not confuse the scalar/vector nature of a function with the nestedness of the argument and return types. A scalar function may operate on or return `LIST` types, for example, and still be called scalar in this context. Scalar vs. vector refers to whether the function implicitly operates on/returns just one instance of the specified data types, or multiple of them.
+
+## Arguments
+
+When a function is used in a plan, it is bound to an argument pack, consisting of zero or more individual arguments. Currently, all arguments are positionally matched and mandatory; that is, the number of arguments bound to a function must match the number of argument slots in the declaration exactly, except when the last argument slot is variadic.
+
+There are four main types of arguments: value arguments, type arguments, required enumerations, and optional enumerations. These are described in detail in the following subsections.
+
+### Value Arguments
+
+Value arguments are arguments that refer to a data value. These could be constants (literal expressions defined in the plan) or variables (a reference expression that references data being processed by the plan). This is the most common type of argument. The data type expected by the argument is defined using a [data type pattern](#data-type-patterns).
+
+| Property | Description                                     | Required |
+|----------|-------------------------------------------------|----------|
+| Metadata | See [common metadata](#common-metadata).        | Optional |
+| Pattern  | A [data type pattern](#data-type-patterns).     | Required |
+| Constant | Whether this argument is required to be a constant for invocation. For example, in some system a regular expression pattern would only be accepted as a literal and not a column value reference. | Optional, defaults to false |
+
+!!! note
+
+    The *value* of the argument is not usable for return type derivations and constraints, even for constant value arguments. For example, it is not currently possible to assert that a particular function only accepts `i8` values between 0 and 100, nor is it possible to declare a function like `new_zero_decimal(P: i8, S: i8) -> DECIMAL<P,S>`.
+
+!!! note
+
+    Constant value arguments effectively always act like scalars, even for function types that take vectors as input. For example, an aggregate function like `count_matching(regex: const string, values: string) -> i64` would always only need to compile the passed regular expression once, while `count_matching(regex: string, values: string) -> i64` is expected to work even if the regex varies from record to record.
```

Comment:
No. I don't think I should mention that either, because Substrait just doesn't conceptually support this sort of thing right now. The current docs just say const value arguments can only be bound to a literal, and that's it.